gobblin-data-management/src/test/java/org/apache/gobblin/util/commit/CreateDirectoryWithPermissionsCommitStepTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.util.commit;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.Lists;
+import com.google.common.io.Files;
+
+import org.apache.gobblin.commit.CommitStep;
+import org.apache.gobblin.util.filesystem.OwnerAndPermission;
+
+
+/**
+ * Test for {@link CreateDirectoryWithPermissionsCommitStep}.
+ */
+@Test(groups = { "gobblin.commit" })
+public class CreateDirectoryWithPermissionsCommitStepTest {
```

Comment:
since so much similarity, would it make sense to have a commit step test base class, shared between the two, which might avoid repetition while also better illustrating the different semantics between the two commit steps